static/src/javascripts/projects/common/modules/navigation/sticky.js

```diff
@@ -244,44 +246,88 @@ define([
         }.bind(this));
     };
 
+    StickyHeader.prototype.fixedBannerMobile = function (headerTop, bannerHeight) {
+        fastdom.write(function () {
+            this.$els.header.css({
+                position: 'fixed',
+                top: headerTop,
+                width: '100%',
+                'z-index': '1001',
+                'margin-top': 0
+            });
+            this.$els.bannerMobile.css({
+                position: 'fixed',
+                top: this.headerBigHeight + headerTop,
+                width: '100%',
+                'z-index': '999' // Sticky z-index -1 as it should be sticky but should go below the sticky header
+            });
+            this.$els.main.css('margin-top', this.headerBigHeight + bannerHeight);
+        }.bind(this));
+    };
+
     StickyHeader.prototype.updatePositionMobile = function () {
         fastdom.read(function () {
-            var bannerHeight = this.$els.bannerMobile.dim().height,
-                scrollY      = this.$els.window.scrollTop();
+            var bannerHeight      = this.$els.bannerMobile.dim().height,
+                scrollY           = this.$els.window.scrollTop(),
+                smartBannerHeight = smartAppBanner.getMessageHeight();
 
             fastdom.write(function () {
                 this.setScrollDirection(scrollY);
 
-                //header, navigation and banner are sticky from the beginning
-                if (scrollY < this.config.thresholdMobile) {
-                    fastdom.write(function () {
-                        this.$els.header.css({
-                            position: 'fixed',
-                            top: 0,
-                            width: '100%',
-                            'z-index': '1001',
-                            'margin-top': 0
-                        });
-                        this.$els.bannerMobile.css({
-                            position:  'fixed',
-                            top:       this.headerBigHeight,
-                            width:     '100%',
-                            'z-index': '999' // Sticky z-index -1 as it should be sticky but should go below the sticky header
-                        });
-                        this.$els.main.css('margin-top', this.headerBigHeight + bannerHeight);
-                    }.bind(this));
-                    // Put navigation to its default state
-                    this.setNavigationDefault();
+                if (smartAppBanner.isMessageShown()) {
+                    if (scrollY < smartBannerHeight) {
+                        fastdom.write(function () {
+                            this.$els.header.css({
```

Comment:
would it be better to put the CSS in with the rest of the css, and just add/remove classes?